openassessment/xblock/static/js/spec/lms/oa_staff_area.js

```diff
@@ -0,0 +1,302 @@
+
+            // Expect that the server was instructed to reschedule Unifinished Taks
+            expect(server.rescheduleUnfinishedTasks).toHaveBeenCalled();
+        });
+    });
+
+    describe('Submission Management', function() {
+        it('updates submission cancellation button when comments changes', function() {
+            // Prevent the server's response from resolving,
+            // so we can see what happens before view gets re-rendered.
+            spyOn(server, 'cancelSubmission').and.callFake(function() {
+                return $.Deferred(function() {}).promise();
+            });
+
+            // Load the fixture
+            loadFixtures('oa_student_info.html');
+
+            var view = createStaffArea();
+
+            // comments is blank --> cancel submission button disabled
+            view.comment('');
+            view.handleCommentChanged();
+            expect(view.cancelSubmissionEnabled()).toBe(false);
+
+            // Response is whitespace --> cancel submission button disabled
+            view.comment('               \n      \n      ');
+            view.handleCommentChanged();
+            expect(view.cancelSubmissionEnabled()).toBe(false);
+
+            // Response is not blank --> cancel submission button enabled
+            view.comment('Cancellation reason.');
+            view.handleCommentChanged();
+            expect(view.cancelSubmissionEnabled()).toBe(true);
+        });
+
+        it('submits the cancel submission comments to the server', function() {
+            spyOn(server, 'cancelSubmission').and.callThrough();
+
+            // Load the fixture
+            loadFixtures('oa_student_info.html');
+            var view = createStaffArea();
+
+            view.comment('Cancellation reason.');
+            view.cancelSubmission('Bob');
+
+            expect(server.cancelSubmission).toHaveBeenCalledWith('Bob', 'Cancellation reason.');
+        });
+    });
+
+    describe('Staff Toolbar', function() {
+        beforeEach(function() {
+            loadFixtures('oa_base_course_staff.html');
+        });
+
+        var getVisibleStaffPanels = function(view) {
+            return $('.wrapper--ui-staff', view.element).not('.is--hidden');
+        };
+
+        it('shows the correct buttons with no panels initially', function() {
+            var view = createStaffArea(),
+                $buttons = $('.ui-staff__button', view.element);
+            expect($buttons.length).toBe(2);
+            expect($($buttons[0]).text().trim()).toEqual('Staff Tools');
+            expect($($buttons[1]).text().trim()).toEqual('Staff Info');
+            expect(getVisibleStaffPanels(view).length).toBe(0);
+        });
+
+        it('shows the "Staff Tools" panel when the button is clicked', function() {
+            var view = createStaffArea(),
+                $staffToolsButton = $('.button-staff-tools', view.element),
+                $visiblePanels;
+            expect($staffToolsButton.length).toBe(1);
```

Comment:
Nit: you could remove a lot of the duplication in these test points by creating helper methods where you pass in the button you are working with. Personally, I would find that easier to read and maintain.